test/execution/test_plan.py

```diff
@@ -708,3 +709,38 @@ def plan_test_2(x, y):
     pointer_to_result = pointer_plan(pointer_to_data_1, pointer_to_data_2)
     result = pointer_to_result.get()
     assert (result == x12).all
+
+
+def test_training_plan_can_be_traced(hook, workers):
+    # sy.ID_PROVIDER = list(reversed(range(1, 10000)))
+
+    @sy.func2plan(args_shape=[(5, 5)])
+    def autograd_test(X):
+        y = X * 5
+        y = -y.log() / 2
+        y = y.sum()
+        y.backward()
+        return X.grad
+
+    X = th.ones(5, 5, requires_grad=True)
+
+    # Result of torch autograd
+    torch_grads = autograd_test(X)
+
+    # Result of traced backprop
+    autograd_test.forward = None
+    plan_grads = autograd_test(X)
+
+    # Translate to torchscript
+    autograd_test.add_translation(PlanTranslatorTorchscript)
+
+    # Result of torchscript'ed traced backprop
+    ts_plan_grads = autograd_test.torchscript(X)
+
+    # (debug out)
+    print("Traced Plan:\n", autograd_test.code)
+    print("Torchscript Plan:\n", autograd_test.torchscript.code)
+
+    # Test all results are equal
+    assert torch_grads.eq(plan_grads).all()
+    assert torch_grads.eq(ts_plan_grads).all()
```

Comment:
This seems like two tests rolled into one. The Torchscript translation part should probably go to `test_translation.py`.